### DISCUSSION TITLE
Learning parameters to train with stress, forces and energy

### DISCUSSION THREAD

**Discussion by mesonepigreco** [mir-group/nequip#233] [Q&A]
Dear developers,
I am trying to train a nequip potential with data with different cell shape and a different number of atoms.
I followed the example files provided, however I am getting not very good values of errors in energies and forces and I am worried I'm missing some of the meaning of the hyper parameters that could maybe be optimized better:

For example, I'm using a loss_coeffs of:
![loss](https://user-images.githubusercontent.com/28493524/181736839-fe927ba6-cf66-48bf-b5e9-337a1bc67bf9.png)

What is the meaning of these parameters? how can they be optimized? 

Also, this is how the validation plots look (taken form wandb.ai, validation energy/N_mae and validation_forces_mae on the two atomic species; units are  eV and A). I see a lot of sharp peaks toward high values. it seems there s an overall trend to go down, but seems there are this spikes that are hampering getting to very good precision. Is it normal or it is a signature of something wrong happening with my dataset/parameters? 
  
![energy](https://user-images.githubusercontent.com/28493524/181737608-25e50258-4d6e-466d-bf3d-6d500d469e70.png)

![forces](https://user-images.githubusercontent.com/28493524/181737766-039b1635-9e37-4a5e-ad62-41544b736e64.png)

Thanks for the attention,
Best regards, 
Lorenzo Monacelli

**Comment by Linux-cpp-lisp** [comment]
HI @mesonepigreco ,

> For example, I'm using a loss_coeffs of:

Was this problem introduced by adding stress to your training, or this is just an issue you are facing in general?

Without knowing more, I would suggest trying a lower learning rate... spikes like this can be a sign of the training "blowing up" when it takes too large a step in weight space.

**Reply by mesonepigreco** [reply]
Hi Alby,
Thanks  for the response,
This was a general issue I was facing even before introducing the stress. I am now trying to lower the learning rate (from 0.005 to 0.001) and keep you updated if the issue is solved.
Cheers,
Lorenzo
